src/dns/aardvark.rs

```diff
@@ -0,0 +1,256 @@
+                                            let mut network_gateway_v6: String = "".to_string();
+                                            let mut container_ip_v4: String = "".to_string();
+                                            let mut container_ip_v6: String = "".to_string();
+                                            let container_ip = subnet.ipnet.addr();
+                                            let gateway = match subnet.gateway {
+                                                Some(gateway) => gateway,
+                                                None => IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
+                                            };
+
+                                            if !gateway.is_unspecified() {
+                                                if gateway.is_ipv4() {
+                                                    network_gateway_v4 = gateway.to_string();
+                                                } else {
+                                                    network_gateway_v6 = gateway.to_string();
+                                                }
+                                            }
+
+                                            if container_ip.is_ipv4() {
+                                                container_ip_v4 = container_ip.to_string();
+                                            } else {
+                                                container_ip_v6 = container_ip.to_string();
+                                            }
+
+                                            result.push(AardvarkEntry {
+                                                network_name: network_name.clone(),
+                                                network_gateway_v4,
+                                                network_gateway_v6,
+                                                container_id: container_id.clone(),
+                                                container_ip_v6,
+                                                container_ip_v4,
+                                                container_name: Vec::from([container_name.clone()]),
+                                            });
+                                        }
+                                    }
+                                    None => continue,
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        result
+    }
+
+    pub fn delete_entry(self, container_id: String, network_name: String) -> Result<()> {
+        let path = Path::new(&self.config).join(network_name);
+        let mut file = File::open(&path)?;
+        let reader = BufReader::new(&mut file);
+
+        let lines: Vec<_> = reader
+            .lines()
+            .map(|l| l.expect("Couldn't read a line"))
+            .collect();
```

Comment:
I think this reads the full file into memory, I think we should only read line by line.

```
while read line
  if not contains ID
    write line to new file
```

Reply:
Removing entries have lot of trade-offs. We already do lot of `i/o` while writing, I think doing `i/o` for each line while reading could become we can avoid it since in most of the cases `lines` would be few so memory might not become a bottle-neck.

But if it becomes bottleneck we can optimize this later.

Reply:
But your code already reads line by line into an vector. Please do not read everything into memory the amount of reads will still be the same.

Reply:
fixed now reading file at once.